src/distilabel/utils/serialization_v2.py

```diff
@@ -0,0 +1,256 @@
+# Copyright 2023-present, Argilla, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import importlib
+import json
+import os
+from pathlib import Path
+from typing import Any, Dict, Generic, List, Literal, Optional, Type, TypeVar, get_args
+
+import yaml
+from pydantic import BaseModel
+
+T = TypeVar("T")
+
+DISTILABEL_FILENAME = "distilabel-file.json"
+
+
+SaveFormat = Literal["json", "yaml"]
+
+
+def _get_class(module: str = None, name: str = None) -> Type:
+    mod = importlib.import_module(module)
+    return getattr(mod, name)
+
+
+def load_from_dict(class_: Dict[str, Any]) -> Generic[T]:
+    """Reads a template (a class serialized) and returns the instance
+    contained.
+
+    Args:
+        template (Dict[str, Any]): Dict containing the template, the dict serialized.
```

Comment:
Should `class_` be renamed to `template` or the other way around?

```suggestion
def load_from_dict(class_: Dict[str, Any]) -> Generic[T]:
    """Reads a template (a class serialized) and returns the instance
    contained.

    Args:
        class_ (Dict[str, Any]): Dict containing the template, the dict serialized.
```